server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -541,6 +557,54 @@ private ClusterState applyCreateIndexWithTemporaryService(
         });
     }
 
+    Template applyContext(
+        CreateIndexClusterStateUpdateRequest request,
+        ClusterState currentState,
+        List<Map<String, Object>> mappings,
+        Settings.Builder settingsBuilder
+    ) throws IOException {
+        if (request.context() != null) {
+            ComponentTemplate componentTemplate = MetadataIndexTemplateService.findComponentTemplate(
+                currentState.metadata(),
+                request.context()
+            );
+
+            if (componentTemplate.template().mappings() != null) {
+                // Mappings added at last (priority to mappings provided)
```

Comment:
Mapping added at last - Are the mappings applied in the reverse order?

Reply:
Yes. https://github.com/opensearch-project/OpenSearch/blob/2b843051a4e71d79fd8fb94014cc8a41a0f97560/server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java#L1316-L1321